example/demo/src/inventory/ui.tsx

```diff
@@ -173,142 +175,91 @@ export function getHomeScreen(): ServerDataState {
           },
-          {
-            $: 'component',
-            component: 'Paragraph',
-            key: 'description',
-            children: {
-              $: 'ref',
-              ref: [`inventory-items`, item.key, 'description'],
-            },
-          },
-          {
-            $: 'component',
-            component: 'XStack',
-            key: 'adjustments',
-            props: {
-              gap: '$3',
-              alignItems: 'center',
-            },
-            children: [
-              {
-                $: 'component',
-                key: 'quantity',
-                component: 'XStack',
-                props: {
-                  gap: '$2',
-                },
-                children: [
-                  {
-                    $: 'component',
-                    key: 'label',
-                    component: 'SizableText',
-                    props: {
-                      size: '$5',
-                    },
-                    children: `Quantity:`,
-                  },
-                  {
-                    $: 'component',
-                    key: 'value',
-                    component: 'SizableText',
-                    props: {
-                      size: '$5',
-                    },
-                    children: {
-                      $: 'ref',
-                      ref: [`inventory-items`, item.key, 'quantity'],
-                    },
-                  },
-                ],
-              },
-              {
-                $: 'component',
-                component: 'Button',
-                key: 'decrement',
-                props: {
-                  theme: 'red',
-                  onPress: handler(() => {
-                    ctx.update(`inventory-items`, (data: Record<string, Item>) => ({
-                      ...data,
-                      [item.key]: {
-                        ...data[item.key],
-                        quantity: data[item.key]!.quantity - 1,
-                      },
-                    }))
-                  }),
+        }}
+        style={{
+          width: '100%',
+          height: 200,
+          backgroundColor: 'white',
+        }}
+        resizeMode="contain"
+      />
+      <YStack key="info" paddingHorizontal="$4" gap="$3">
+        <H2
+          key="title"
+          children={{
+            $: 'ref',
+            ref: [`inventory-items`, item.key, 'title'],
+          }}
+        />
+        <Paragraph
+          key="description"
+          children={{
+            $: 'ref',
+            ref: [`inventory-items`, item.key, 'description'],
+          }}
+        />
+        <XStack key="adjustments" gap="$3" alignItems="center">
+          <XStack key="quantity" gap="$2">
+            <SizableText size="$5">Quantity:</SizableText>
+            <SizableText
+              size="$5"
+              children={{
+                $: 'ref',
+                ref: [`inventory-items`, item.key, 'quantity'],
```

Comment:
We may need a <Ref /> component here so it can be passed more common way, rather than object.